gnovm/pkg/gnolang/preprocess.go

```diff
@@ -2509,6 +2346,206 @@ func preprocess1(store Store, ctx BlockNode, n Node) Node {
 	return nn
 }
 
+// This func aims at syncing logic between op define (:=) and declare(var/const)
+// This will define or declare the variables
+func defineOrDecl(
+	sts []Type,
+	tvs []TypedValue,
+	bn BlockNode,
+	nameExprs []NameExpr,
+	isConst bool,
+) {
+	node := bn
+	if fn, ok := bn.(*FileNode); ok {
+		node = fn.GetParentNode(nil).(*PackageNode)
+	}
+
+	for i := 0; i < len(sts); i++ {
+		nx := nameExprs[i]
+		if nx.Name == blankIdentifier {
+			nx.Path = NewValuePathBlock(0, 0, blankIdentifier)
+		} else {
+			node.Define2(isConst, nx.Name, sts[i], tvs[i])
+			nx.Path = bn.GetPathForName(nil, nx.Name)
+		}
+	}
+}
+
+// Generate the placeholders for types, typeValues, nameExprs for all vars
+func prepareTypeVals(n Node, numNames int) ([]Type, []TypedValue, NameExprs) {
+	if numNames == 0 {
+		panic("cannot assign to 0 name")
+	}
```

Comment:
```suggestion
	if numNames < 1 {
		panic("must have at least one name to assign")
	}
```
I would do this to handle all cases even if it should not happen in theory or use uint as parameter would make sense